applications/video_streaming_demo_enhanced/CMakeLists.txt

```diff
@@ -0,0 +1,236 @@
+             SERVER_PID=\$!
+
+             # Wait for server to start and check logs
+             echo 'Waiting for server to initialize...'
+             sleep 10
+
+             # Verify server started correctly
+             if ! kill -0 \$SERVER_PID 2>/dev/null; then
+               echo '✗ Server process died during startup'
+               cat \$SERVER_LOG
+               rm -f \$SERVER_LOG \$CLIENT_LOG
+               exit 1
+             fi
+
+             echo '✓ Server process is running'
+
+             # Start client in background with log capture
+             echo 'Starting streaming client...'
+             ./applications/video_streaming_demo_enhanced/video_streaming_demo_client/cpp/streaming_client_demo_enhanced --config applications/video_streaming_demo_enhanced/video_streaming_demo_client/cpp/streaming_client_demo_replayer.yaml > \$CLIENT_LOG 2>&1 &
+             CLIENT_PID=\$!
+
+             # Let streaming run for 30 seconds
+             echo 'Letting streaming run for 30 seconds...'
+             sleep 30
+
+             # Stop client and server gracefully
+             echo 'Stopping client...'
+             kill -TERM \$CLIENT_PID 2>/dev/null || true
+             sleep 2
+             kill -KILL \$CLIENT_PID 2>/dev/null || true
+
+             echo 'Stopping server...'
+             kill -TERM \$SERVER_PID 2>/dev/null || true
+             sleep 2
+             kill -KILL \$SERVER_PID 2>/dev/null || true
+
+             # Now verify logs for actual streaming behavior
+             echo ''
+             echo '=== Verifying Server Logs ==='
+             
+             # Check for client connection
+             if grep -q 'Client connected' \$SERVER_LOG; then
+               echo '✓ Server: Client connected'
+               SERVER_SUCCESS=\$((SERVER_SUCCESS + 1))
+             else
+               echo '✗ Server: No client connection detected'
+             fi
+             
+             # Check for upstream connection
+             if grep -q 'Upstream connection established' \$SERVER_LOG; then
+               echo '✓ Server: Upstream connection established'
+               SERVER_SUCCESS=\$((SERVER_SUCCESS + 1))
+             else
+               echo '✗ Server: Upstream connection not established'
+             fi
+             
+             # Check for downstream connection
+             if grep -q 'Downstream connection established' \$SERVER_LOG; then
+               echo '✓ Server: Downstream connection established'
+               SERVER_SUCCESS=\$((SERVER_SUCCESS + 1))
+             else
+               echo '✗ Server: Downstream connection not established'
+             fi
+             
+             # Check for frame processing
+             if grep -q 'Processing UNIQUE frame' \$SERVER_LOG; then
```

Comment:
This check looks good and verifies that the `StreamingServerUpstreamOp::compute` method has run.

Please also add a regex check to verify that frames are handled in `StreamingServerDownstreamOp`.